views.py

```diff
@@ -0,0 +1,169 @@
+def render_cart():
+    flag = 0
+    form = OrderForm()
+    cart = session.get("cart", [])
+    use_it = set()
+    new_cart = []
+    for item in cart:
+        if item not in use_it:
+            new_cart.append(item)
+            use_it.add(item)
+    user = None
+    if session.get("user_id"):
+        user = User.query.filter(User.id == session.get("user_id")).first()
+    session["word"] = correct_word(len(session.get("cart", [])))
+    return render_template("cart.html", form=form, new_cart=new_cart, flag=flag, user=user)
+
+
+@app.route("/add/<item>/")
+def add_to_cart(item):
+    flag = 0
+    form = OrderForm()
+    items = db.session.query(Dish)
+    cart = session.get("cart", [])
+    cart.append(item)
+    session["cart"] = cart
+    for i in items:
+        if i.title == item:
+            if not session.get("price"):
+                session["price"] = 0
+            session["price"] += i.price
+    cart = session.get("cart", [])
+    use_it = set()
+    new_cart = []
+    for item in cart:
+        if item not in use_it:
+            new_cart.append(item)
+            use_it.add(item)
+    user = None
+    if session.get("user_id"):
+        user = User.query.filter(User.id == session.get("user_id")).first()
+    session["word"] = correct_word(len(session.get("cart", [])))
+    return render_template("cart.html", form=form, flag=flag, new_cart=new_cart, user=user)
+
+
+@app.route("/pop/<item>/")
+def pop_from_cart(item):
+    flag = 1
+    form = OrderForm()
+    items = db.session.query(Dish)
+    cart = session.get("cart", [])
+    cart.remove(item)
+    session["cart"] = cart
+    for i in items:
+        if i.title == item:
+            session["price"] -= i.price
+    cart = session.get("cart", [])
+    use_it = set()
+    new_cart = []
+    for item in cart:
+        if item not in use_it:
+            new_cart.append(item)
+            use_it.add(item)
+    user = None
+    if session.get("user_id"):
+        user = User.query.filter(User.id == session.get("user_id")).first()
+    session["word"] = correct_word(len(session.get("cart", [])))
+    return render_template("cart.html", form=form, flag=flag, new_cart=new_cart, user=user)
+
+
+@app.route("/ordered/", methods=["GET", "POST"])
+def render_order():
+    form = OrderForm()
+    if request.method == "POST":
+        id_o = len(db.session.query(User).get(session["user_id"])) + 1
+        date = datetime.now().date()
+        mail = form.mail.data
+        name = form.name.data
+        address = form.address.data
+        user_id = session["user_id"]
+        sum_price = session["price"]
+        dishes = session.get("cart", [])
+        order = Order(id=id_o, date=date, mail=mail, name=name, address=address, user_id=user_id, sum_price=sum_price,
+                      dishes=dishes)
+        db.session.add(order)
+        db.session.commit()
+        return render_template("ordered.html")
+    else:
+        orders = db.session.query(Order)
```

Comment:
take all orders?! maybe filter by user?

Reply:
They are filtered by user)
The order table has a user column

Reply:
But once again, you just take whole table from db!